pkg/controllers/plugin/util.go

```diff
@@ -53,3 +62,127 @@ func convertRuntimeObjectToCoreV1Service(o interface{}) (*corev1.Service, error)
 		return nil, fmt.Errorf("unsupported runtime.Object type: %T", obj)
 	}
 }
+
+func isPayloadReadyRunning(o interface{}) bool {
+	switch obj := o.(type) {
+	case *appsv1.Deployment:
+		if obj.Status.ReadyReplicas == obj.Status.Replicas && obj.Status.ReadyReplicas == obj.Status.AvailableReplicas {
+			return true
+		}
+	case *appsv1.StatefulSet:
+		if obj.Status.ReadyReplicas == obj.Status.Replicas && obj.Status.ReadyReplicas == obj.Status.AvailableReplicas {
+			return true
+		}
+	case *appsv1.DaemonSet:
+		if obj.Status.NumberReady == obj.Status.DesiredNumberScheduled && obj.Status.NumberReady == obj.Status.NumberAvailable {
+			return true
+		}
+	case *appsv1.ReplicaSet:
+		if obj.Status.ReadyReplicas == obj.Status.Replicas && obj.Status.ReadyReplicas == obj.Status.AvailableReplicas {
+			return true
+		}
+	case *batchv1.CronJob:
+		// CronJob does not have a status field just for the job, so we need to check the last successful time
+		if obj.Status.LastSuccessfulTime != nil {
```

Comment:
This does not necessarily mean that the last run was successful. Rather making sure lastScheduleTime equals the lastSuccessfulTime

```yaml
status:
  lastScheduleTime: "2024-08-08T01:00:00Z"
  lastSuccessfulTime: "2024-08-08T01:00:05Z"
```